src/tcp-intercept.cxx

```diff
@@ -273,10 +279,15 @@ static void listening_socket_ready_for_read(EV_P_ ev_io *w, int revents) {
 		LogInfo(_("%1$s: Connection intercepted"), new_con->id.c_str());
 
 		new_con->s_server = Socket::socket(server_addr->addr_family(), SOCK_STREAM, 0);
-		val = 1;
-		new_con->s_server.setsockopt(IPPROTO_TCP, TCP_NODELAY, (char *) &val, sizeof(val));
-		val = 1;
-		new_con->s_server.setsockopt(SOL_SOCKET, SO_KEEPALIVE, &val, sizeof(val));
+		if(KEEPALIVE){
+			int val = 1;
+			new_con->s_server.setsockopt(SOL_SOCKET, SO_KEEPALIVE, &val, sizeof(val));
+			//LogInfo(_("%1$s: Enabling keepalive on s_server"), new_con->id.c_str());		
+		}
+		if(NODELAY){
```

Comment:
Very minor remark:
In the client section, you first do nodelay, then keepalive; Here, you do it the other way around... It would look better if these orders where the same. (But technically, this does not matter)